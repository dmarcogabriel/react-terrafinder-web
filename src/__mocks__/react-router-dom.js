const originalModule = jest.requireActual('react-router-dom');

module.exports = {
  ...originalModule,
  useHistory: () => ({
    history: { push: jest.fn() },
  }),
};
