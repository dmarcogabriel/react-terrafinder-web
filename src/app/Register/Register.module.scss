@import 'styles';

$padding: 1.5rem;

.container {
  display: flex;
  flex-direction: column;

  @media ($desktop) {
    flex-direction: row;
  }

  .welcome {
    background-image: url('./background.jpg');
    background-size: cover;
    height: 100vh;
    background-repeat: no-repeat;
    padding: $padding;

    @media ($desktop) {
      width: 50%;
      background-size: cover;
    }

    > h1 {
      font-size: 5rem;
      color: $gray-light;

      @media ($desktop) {
        margin: 2rem 3rem;
      }
    }
  }

  .register {
    padding: $padding;

    @media ($desktop) {
      width: 50%;
    }

    .form {
      @media ($desktop) {
        padding: 2rem 5rem;
      }

      > h2 {
        color: $primary-dark;
        font-size: 2rem;
        margin-bottom: 1rem;
        text-align: center;
      }

      > p {
        color: $gray-light;
        text-align: center;
      }

      .groupedInputs {
        @media ($desktop) {
          display: flex;
          justify-content: space-between;
          align-items: center;
        }
      }

      .registerButton {
        background: $green;
        width: 100%;
        text-align: center;
        margin: 1rem 0 2.5rem;
      }
    }
  }
}
