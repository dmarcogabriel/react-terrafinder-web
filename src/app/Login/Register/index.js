import React from 'react';
import { Link, useHistory } from 'react-router-dom';
import Button from 'common/components/atm/Button';
import Input from 'common/components/Input';
import { useFormik } from 'formik';
import { object, string } from 'yup';
import api from 'services/api';
import { useNotification, NOTIFICATION_TYPES } from 'hooks/useNotification';
import {
  form,
  groupedInputs,
  actionButton,
} from '../components/Login.module.scss';
import { LoginContainer } from '../components';

export const Register = () => {
  const history = useHistory();
  const { showNotification } = useNotification();

  const { values, errors, handleChange, handleSubmit } = useFormik({
    initialValues: {
      firstName: '',
      lastName: '',
      email: '',
      cpf: '',
      phone: '',
      password: '',
    },
    validationSchema: object().shape({
      firstName: string().required('Campo obrigatório!'),
      lastName: string().required('Campo obrigatório!'),
      email: string().email().required('Campo obrigatório!'),
      cpf: string().required('Campo obrigatório!'),
      phone: string().required('Campo obrigatório!'),
      password: string()
        .min(6, 'Mínimo 6 caracteres')
        .required('Campo obrigatório!'),
    }),
    async onSubmit() {
      try {
        await api.post('users', { ...values });

        showNotification(
          'Cadastro efetuado com sucesso',
          NOTIFICATION_TYPES.SUCCESS
        );
        history.push('/login');
      } catch (error) {
        showNotification('Erro ao efetuar cadastro', NOTIFICATION_TYPES.ERROR);
      }
    },
  });

  return (
    <LoginContainer>
      <div className={form}>
        <h2>Cadastre-se para cadastrar seu anúncio</h2>

        <p>
          Já tem uma conta?
          <Link to="/login"> Faça o login aqui</Link>
        </p>

        <div className={groupedInputs}>
          <Input
            dataTestId="firstNameInput"
            label="Primeiro nome"
            value={values.firstName}
            onChange={handleChange('firstName')}
            errorMessage={errors.firstName}
          />

          <Input
            dataTestId="lastNameInput"
            label="Sobrenome"
            value={values.lastName}
            onChange={handleChange('lastName')}
            errorMessage={errors.lastName}
          />
        </div>

        <Input
          dataTestId="emailInput"
          label="E-mail"
          type="email"
          value={values.email}
          onChange={handleChange('email')}
          errorMessage={errors.email}
        />

        <div className={groupedInputs}>
          <Input
            dataTestId="phoneInput"
            label="Telefone para contato"
            value={values.phone}
            onChange={handleChange('phone')}
            errorMessage={errors.phone}
          />

          <Input
            dataTestId="cpfInput"
            label="CPF"
            value={values.cpf}
            onChange={handleChange('cpf')}
            errorMessage={errors.cpf}
          />
        </div>

        <Input
          dataTestId="passInput"
          label="Senha"
          type="password"
          value={values.password}
          errorMessage={errors.password}
          onChange={handleChange('password')}
        />

        <Button
          dataTestId="registerButton"
          className={actionButton}
          onClick={handleSubmit}
        >
          Cadastrar
        </Button>

        <p>
          Se cadastrando, você concorda com nossos
          <Link to="/">Termos de Uso</Link>
          {' e '}
          <Link to="/">Política de privacidade</Link>
        </p>
      </div>
    </LoginContainer>
  );
};
