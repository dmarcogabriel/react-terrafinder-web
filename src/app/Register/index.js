import React from 'react';
import { Link } from 'react-router-dom';
import Button from 'common/Button';
import Input from 'common/Input';
import {
  container,
  welcome,
  register,
  registerButton,
  form,
  groupedInputs,
} from './Register.module.scss';

export default function Register() {
  return (
    <div className={container}>
      <div className={welcome}>
        <h1>Bem vindo!</h1>
      </div>

      <div className={register}>
        <div className={form}>
          <h2>Cadastre-se para cadastrar seu anúncio</h2>

          <p>
            Já tem uma conta?
            <Link to="/"> Faça o login aqui</Link>
          </p>

          <div className={groupedInputs}>
            <div className="input">
              <Input label="Primeiro nome" onChange={console.log} />
            </div>

            <div className="input">
              <Input label="Sobrenome" onChange={console.log} />
            </div>
          </div>

          <div className="input">
            <Input label="E-mail" onChange={console.log} />
          </div>

          <div className={groupedInputs}>
            <div className="input">
              <Input label="Telefone para contato" onChange={console.log} />
            </div>

            <div className="input">
              <Input label="CPF" onChange={console.log} />
            </div>
          </div>

          <div className="input">
            <Input
              label="Senha"
              type="password"
              errorMessage="Mínimo 6 caracteres"
              onChange={console.log}
            />
          </div>

          <Button className={registerButton}>Cadastrar</Button>

          <p>
            Se cadastrando, você concorda com nossos
            <Link to="/">Termos de Uso</Link>
            {' e '}
            <Link to="/">Política de privacidade</Link>
          </p>
        </div>
      </div>
    </div>
  );
}
