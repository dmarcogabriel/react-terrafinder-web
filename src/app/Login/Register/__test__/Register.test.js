import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
// import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Register from '..';

const Component = () => (
  <BrowserRouter>
    <Register />
  </BrowserRouter>
);

describe('<Register />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Component />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // Add tests here...

  it('matches snapshot', () => {
    const tree = renderer.create(<Component />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
