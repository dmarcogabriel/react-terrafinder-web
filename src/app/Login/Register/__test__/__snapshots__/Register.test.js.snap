// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> matches snapshot 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidth100% MuiContainer-disableGutters css-ceeg19-MuiContainer-root"
>
  <div
    className="MuiBox-root css-11mpnc4"
  >
    <h1
      className="MuiTypography-root MuiTypography-h2 css-dfmhkl-MuiTypography-root"
    >
      Bem vindo!
    </h1>
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 css-d3mjh9-MuiPaper-root"
  >
    <div
      className="MuiBox-root css-cacgvk"
    >
      <h2
        className="MuiTypography-root MuiTypography-h4 css-7kg1mh-MuiTypography-root"
      >
        Cadastre-se para criar seus anúncios
      </h2>
      <p
        className="MuiTypography-root MuiTypography-body1 css-d3ow91-MuiTypography-root"
      >
        Já possui uma conta?
         
        <a
          href="/login"
        >
          Faça o login aqui
        </a>
      </p>
    </div>
    <div
      className="MuiBox-root css-1l9sgno"
    >
      <div
        className="MuiBox-root css-3lvcri"
      >
        <div
          className="MuiBox-root css-1663l76"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
          >
            Primeiro nome
          </label>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeSmall css-1ucxkcv-MuiSvgIcon-root"
            data-testid="MoodIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm3.5-9c.83 0 1.5-.67 1.5-1.5S16.33 8 15.5 8 14 8.67 14 9.5s.67 1.5 1.5 1.5zm-7 0c.83 0 1.5-.67 1.5-1.5S9.33 8 8.5 8 7 8.67 7 9.5 7.67 11 8.5 11zm3.5 6.5c2.33 0 4.31-1.46 5.11-3.5H6.89c.8 2.04 2.78 3.5 5.11 3.5z"
            />
          </svg>
        </div>
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="firstNameInput"
              disabled={false}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1a1ihhj"
              >
                <span
                  className="notranslate"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root css-c8l3hs"
      >
        <div
          className="MuiBox-root css-1663l76"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
          >
            Sobrenome
          </label>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeSmall css-1ucxkcv-MuiSvgIcon-root"
            data-testid="MoodIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm3.5-9c.83 0 1.5-.67 1.5-1.5S16.33 8 15.5 8 14 8.67 14 9.5s.67 1.5 1.5 1.5zm-7 0c.83 0 1.5-.67 1.5-1.5S9.33 8 8.5 8 7 8.67 7 9.5 7.67 11 8.5 11zm3.5 6.5c2.33 0 4.31-1.46 5.11-3.5H6.89c.8 2.04 2.78 3.5 5.11 3.5z"
            />
          </svg>
        </div>
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="lastNameInput"
              disabled={false}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1a1ihhj"
              >
                <span
                  className="notranslate"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiBox-root css-1ebprri"
    >
      <div
        className="MuiBox-root css-1663l76"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
        >
          E-mail
        </label>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeSmall css-1ucxkcv-MuiSvgIcon-root"
          data-testid="MoodIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm3.5-9c.83 0 1.5-.67 1.5-1.5S16.33 8 15.5 8 14 8.67 14 9.5s.67 1.5 1.5 1.5zm-7 0c.83 0 1.5-.67 1.5-1.5S9.33 8 8.5 8 7 8.67 7 9.5 7.67 11 8.5 11zm3.5 6.5c2.33 0 4.31-1.46 5.11-3.5H6.89c.8 2.04 2.78 3.5 5.11 3.5z"
          />
        </svg>
      </div>
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
            data-testid="emailInput"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="exemplo@email.com.br"
            required={false}
            type="email"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1a1ihhj"
            >
              <span
                className="notranslate"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiBox-root css-1l9sgno"
    >
      <div
        className="MuiBox-root css-3lvcri"
      >
        <div
          className="MuiBox-root css-1663l76"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
          >
            Telefone para contato
          </label>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeSmall css-1ucxkcv-MuiSvgIcon-root"
            data-testid="MoodIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm3.5-9c.83 0 1.5-.67 1.5-1.5S16.33 8 15.5 8 14 8.67 14 9.5s.67 1.5 1.5 1.5zm-7 0c.83 0 1.5-.67 1.5-1.5S9.33 8 8.5 8 7 8.67 7 9.5 7.67 11 8.5 11zm3.5 6.5c2.33 0 4.31-1.46 5.11-3.5H6.89c.8 2.04 2.78 3.5 5.11 3.5z"
            />
          </svg>
        </div>
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="phoneInput"
              disabled={false}
              maxLength={15}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="(14) 99999-8888"
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1a1ihhj"
              >
                <span
                  className="notranslate"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root css-c8l3hs"
      >
        <div
          className="MuiBox-root css-1663l76"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
          >
            CPF
          </label>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeSmall css-1ucxkcv-MuiSvgIcon-root"
            data-testid="MoodIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm3.5-9c.83 0 1.5-.67 1.5-1.5S16.33 8 15.5 8 14 8.67 14 9.5s.67 1.5 1.5 1.5zm-7 0c.83 0 1.5-.67 1.5-1.5S9.33 8 8.5 8 7 8.67 7 9.5 7.67 11 8.5 11zm3.5 6.5c2.33 0 4.31-1.46 5.11-3.5H6.89c.8 2.04 2.78 3.5 5.11 3.5z"
            />
          </svg>
        </div>
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="cpfInput"
              disabled={false}
              maxLength={14}
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="xxx.xxx.xxx-xx"
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1a1ihhj"
              >
                <span
                  className="notranslate"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiBox-root css-1ebprri"
    >
      <div
        className="MuiBox-root css-1663l76"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
        >
          Senha
        </label>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeSmall css-1ucxkcv-MuiSvgIcon-root"
          data-testid="MoodIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm3.5-9c.83 0 1.5-.67 1.5-1.5S16.33 8 15.5 8 14 8.67 14 9.5s.67 1.5 1.5 1.5zm-7 0c.83 0 1.5-.67 1.5-1.5S9.33 8 8.5 8 7 8.67 7 9.5 7.67 11 8.5 11zm3.5 6.5c2.33 0 4.31-1.46 5.11-3.5H6.89c.8 2.04 2.78 3.5 5.11 3.5z"
          />
        </svg>
      </div>
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
            data-testid="passInput"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="password"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1a1ihhj"
            >
              <span
                className="notranslate"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root css-1h4gk8d-MuiButtonBase-root-MuiButton-root"
      data-testid="registerButton"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Cadastrar
    </button>
    <p
      className="MuiTypography-root MuiTypography-body2 css-1ypfif-MuiTypography-root"
    >
      Se cadastrando, você concorda com nossos
       
      <a
        href="/privacy-policy"
      >
        Termos de uso e política de privacidade
      </a>
    </p>
  </div>
</div>
`;
